Reason: We need to define and use the DEST_INST variable when
executing the `make install`.
GRASS GGIS developer closed a ticket on this with falg: wontfix.
See: https://trac.osgeo.org/grass/ticket/1839 
for further details.

The patch is based on `vince` works:
https://trac.osgeo.org/grass/attachment/ticket/1839/patch-Install_make.diff


--- include/Make/Install.make.1	2019-09-26 13:49:50.551485271 +0200
+++ include/Make/Install.make	2019-09-26 13:13:34.000000000 +0200
@@ -50,30 +50,30 @@ install-check-built:
 	fi
 
 install-check-parent:
-	@ INST_PATH=`dirname $(INST_DIR)`; \
+	@ INST_PATH=`dirname $(DESTDIR)$(INST_DIR)`; \
 	while [ ! -d "$$INST_PATH" ]; do \
 		INST_PATH=`dirname $$INST_PATH`; \
 	done; \
-	if [ ! -d "$(INST_DIR)" -a ! -w "$$INST_PATH" ] ; then \
+	if [ ! -d "$(DESTDIR)$(INST_DIR)" -a ! -w "$$INST_PATH" ] ; then \
 		echo "ERROR: Directory $$INST_PATH is a parent directory of your" >&2 ; \
-		echo "  install directory $(INST_DIR) and is not writable." >&2 ; \
+		echo "  install directory $(DESTDIR)$(INST_DIR) and is not writable." >&2 ; \
 		echo "  Perhaps you need root access." >&2 ; \
 		echo "  Installation aborted, exiting Make." >&2 ; \
 		exit 1; \
 	fi
 
 install-check-writable:
-	@ if [ -d "$(INST_DIR)" -a ! -w "$(INST_DIR)" ] ; then \
-		echo "ERROR: Your install directory $(INST_DIR) is not writable." >&2 ; \
+	@ if [ -d "$(DESTDIR)$(INST_DIR)" -a ! -w "$(DESTDIR)$(INST_DIR)" ] ; then \
+		echo "ERROR: Your install directory $(DESTDIR)$(INST_DIR) is not writable." >&2 ; \
 		echo "  Perhaps you need root access." >&2 ; \
 		echo "  Installation aborted, exiting Make." >&2 ; \
 		exit 1; \
 	fi
 
 install-check-prefix:
-	@ result=`echo "$(INST_DIR)" | awk '{ if ($$1 ~ /grass/) print $$1 }'`; \
+	@ result=`echo "$(DESTDIR)$(INST_DIR)" | awk '{ if ($$1 ~ /grass/) print $$1 }'`; \
 	if [ "$$result" = "" ] ; then \
-		echo "WARNING: Your install directory $(INST_DIR)" >&2 ; \
+		echo "WARNING: Your install directory $(DESTDIR)$(INST_DIR)" >&2 ; \
 		echo "  does not contain the word 'grass'." >&2 ; \
 		echo "  It is highly recommended that the word 'grass' be part" >&2 ; \
 		echo "  of your install directory to avoid conflicts." >&2 ; \
@@ -86,7 +86,7 @@ install-check-prefix:
 	fi
 
 ifneq ($(strip $(MINGW)),)
-STARTUP = $(INST_DIR)/etc/$(GRASS_NAME).py
+STARTUP = $(DESTDIR)$(INST_DIR)/etc/$(GRASS_NAME).py
 else
 STARTUP = $(UNIX_BIN)/$(GRASS_NAME)
 endif
@@ -96,65 +96,67 @@ TMPGISRC = demolocation/.grassrc$(GRASS_
 PLATMAKE = include/Make/Platform.make
 GRASSMAKE = include/Make/Grass.make
 
-real-install: | $(INST_DIR) $(UNIX_BIN)
-	-tar cBCf $(GISBASE) - . | tar xBCf $(INST_DIR) - 2>/dev/null
-	-rm $(INST_DIR)/$(GRASS_NAME).tmp
-	$(MAKE) $(STARTUP)
+real-install: | $(DESTDIR)$(INST_DIR) $(UNIX_BIN)
+	-tar cBCf $(GISBASE) - . | tar xBCf $(DESTDIR)$(INST_DIR) - 2>/dev/null
+	-rm $(DESTDIR)$(INST_DIR)/$(GRASS_NAME).tmp
 
-	-rm $(INST_DIR)/$(FONTCAP)
-	$(MAKE) $(INST_DIR)/$(FONTCAP)
+	$(MAKE_DIR_CMD) -p $(DESTDIR)$(UNIX_BIN)
+	$(MAKE) $(DESTDIR)$(STARTUP)
 
-	-rm $(INST_DIR)/$(TMPGISRC)
-	$(MAKE) $(INST_DIR)/$(TMPGISRC)
+	-rm $(DESTDIR)$(INST_DIR)/$(FONTCAP)
+	$(MAKE) $(DESTDIR)$(INST_DIR)/$(FONTCAP)
 
-	-rm $(INST_DIR)/$(PLATMAKE)
-	$(MAKE) $(INST_DIR)/$(PLATMAKE)
+	-rm $(DESTDIR)$(INST_DIR)/$(TMPGISRC)
+	$(MAKE) $(DESTDIR)$(INST_DIR)/$(TMPGISRC)
 
-	-rm $(INST_DIR)/$(GRASSMAKE)
-	$(MAKE) $(INST_DIR)/$(GRASSMAKE)
+	-rm $(DESTDIR)$(INST_DIR)/$(PLATMAKE)
+	$(MAKE) $(DESTDIR)$(INST_DIR)/$(PLATMAKE)
 
-	-$(INSTALL) config.status $(INST_DIR)/config.status
-	-$(CHMOD) -R a+rX $(INST_DIR) 2>/dev/null
+	-rm $(DESTDIR)$(INST_DIR)/$(GRASSMAKE)
+	$(MAKE) $(DESTDIR)$(INST_DIR)/$(GRASSMAKE)
+
+	-$(INSTALL) config.status $(DESTDIR)$(INST_DIR)/config.status
+	-$(CHMOD) -R a+rX $(DESTDIR)$(INST_DIR) 2>/dev/null
 
 ifneq ($(findstring darwin,$(ARCH)),)
 	@# enable OSX Help Viewer
-	@/bin/ln -sfh "$(INST_DIR)/docs/html" /Library/Documentation/Help/GRASS-$(GRASS_VERSION_MAJOR).$(GRASS_VERSION_MINOR)
+	@/bin/ln -sfh "$(DESTDIR)$(INST_DIR)/docs/html" /Library/Documentation/Help/GRASS-$(GRASS_VERSION_MAJOR).$(GRASS_VERSION_MINOR)
 endif
 
-$(INST_DIR) $(UNIX_BIN):
+$(DESTDIR)$(INST_DIR) $(UNIX_BIN):
 	$(MAKE_DIR_CMD) $@
 
-$(STARTUP): $(ARCH_DISTDIR)/$(GRASS_NAME).tmp
-	sed -e 's#'@GISBASE@'#'$(INST_DIR)'#g' \
+$(DESTDIR)$(STARTUP): $(ARCH_DISTDIR)/$(GRASS_NAME).tmp
+	sed -e 's#'@GISBASE@'#'$(DESTDIR)$(INST_DIR)'#g' \
 	    -e 's#'@LD_LIBRARY_PATH_VAR@'#'$(LD_LIBRARY_PATH_VAR)'#g' \
 	    -e 's#'@CONFIG_PROJSHARE@'#'$(PROJSHARE)'#g' \
 	    $< > $@
 	-$(CHMOD) a+x $@
 
 define fix_gisbase
-sed -e 's#$(GISBASE)#$(INST_DIR)#g' $< > $@
+sed -e 's#$(GISBASE)#$(DESTDIR)$(INST_DIR)#g' $< > $@
 endef
 
 define fix_grass_home
-sed -e 's#^\(GRASS_HOME.[^=]*\).*#\1= $(INST_DIR)#g' \
-    -e 's#$(GISBASE)#$(INST_DIR)#g' $< > $@
+sed -e 's#^\(GRASS_HOME.[^=]*\).*#\1= $(DESTDIR)$(INST_DIR)#g' \
+    -e 's#$(GISBASE)#$(DESTDIR)$(INST_DIR)#g' $< > $@
 endef
 
 define fix_grass_arch
-sed -e 's#^\(ARCH_DISTDIR.[^=]*\).*#\1= $(INST_DIR)#g' \
+sed -e 's#^\(ARCH_DISTDIR.[^=]*\).*#\1= $(DESTDIR)$(INST_DIR)#g' \
     -e 's#^\(ARCH_BINDIR.[^=]*\).*#\1= $(UNIX_BIN)#g' $< > $@
 endef
 
-$(INST_DIR)/$(FONTCAP): $(GISBASE)/$(FONTCAP)
+$(DESTDIR)$(INST_DIR)/$(FONTCAP): $(GISBASE)/$(FONTCAP)
 	$(call fix_gisbase)
 
-$(INST_DIR)/$(TMPGISRC): $(GISBASE)/$(TMPGISRC)
+$(DESTDIR)$(INST_DIR)/$(TMPGISRC): $(GISBASE)/$(TMPGISRC)
 	$(call fix_gisbase)
 
-$(INST_DIR)/$(PLATMAKE): $(GISBASE)/$(PLATMAKE)
+$(DESTDIR)$(INST_DIR)/$(PLATMAKE): $(GISBASE)/$(PLATMAKE)
 	$(call fix_grass_home)
 
-$(INST_DIR)/$(GRASSMAKE): $(GISBASE)/$(GRASSMAKE)
+$(DESTDIR)$(INST_DIR)/$(GRASSMAKE): $(GISBASE)/$(GRASSMAKE)
 	$(call fix_grass_arch)
 
 install-macosx:
@@ -178,9 +180,9 @@ real-bindist:
 	-size=`ls -l $(BINDISTNAME).tar.gz | awk '{print $$5}'`; \
 	    sed -e "s/BIN_DIST_VERSION/$(GRASS_VERSION_NUMBER)-$(ARCH)-$(DATE)/" \
 	    -e "s/GRASSPRG_NAME/$(GRASS_NAME)/" \
-	    -e "s/SIZE_TAR_FILE/$$size/" -e "s#BIN_DIST_DIR#'$(INST_DIR)'#" \
+	    -e "s/SIZE_TAR_FILE/$$size/" -e "s#BIN_DIST_DIR#'$(DESTDIR)$(INST_DIR)'#" \
 	    -e "s/ARCHITECTURE/$(ARCH)/" \
-	    -e "s/LD_LIBRARY_PATH_VAR/$(LD_LIBRARY_PATH_VAR)/" \
+	    -e "s/LD_LIBRARY_PATH_VAR/$(DESTDIR)$(LD_LIBRARY_PATH_VAR)/" \
 	    -e "s/TEST_STR=/TEST_STR=executable/" \
 	    -e "s#IMPORTANT.*#Generated from the binaryInstall.src file using the command make bindist#" \
 	    -e "s/# executable shell.*//" -e "s/# make bindist.*//" \
